### PR TITLE
[Not to be Merged] Rubi progress

### PR DESCRIPTION
**Pull request by ashishkg0022**
Currently I have properly implemented a structure with the following : 
1. Fixed issue of parsetools
2. No lambdas anywhere
3. Generate a single constraint file
4.  Fixed order in which rules are matched. It was the greatest cause of recursion errors. I have kept a uniform numbering of rules. 
5. Generated through parsetools.

### CODE REVIEW

**Comment by ashishkg0022** [comment]
ping @parsoyaarihant  @Upabjojr

File: sympy/integrals/rubi/tests/test_utility_function.py [hunk 1]
```diff
@@ -106,6 +107,7 @@ def test_Coefficient():
     assert Coefficient(a + b*x + c*x**3, x, 0) == a
     assert Coefficient(a + b*x + c*x**3, x, 4) == 0
     assert Coefficient(b*x + c*x**3, x, 3) == c
+    assert Coefficient(x, x, -1) == 0
```
Comment (Abdullahjavednesar) [inline sympy/integrals/rubi/tests/test_utility_function.py]: Wasn't this passing before?

Comment (ashishkg0022) [reply sympy/integrals/rubi/tests/test_utility_function.py]: No , It was causing an exception.

Comment (Abdullahjavednesar) [reply sympy/integrals/rubi/tests/test_utility_function.py]: Well the use of `expand` looks good.

**Pull request closed without merging**